mkall/MKReporter.mm

```diff
@@ -0,0 +1,95 @@
+// Part of Measurement Kit <https://measurement-kit.github.io/>.
+// Measurement Kit is free software under the BSD license. See AUTHORS
+// and LICENSE for more information on the copying conditions.
+
+#import "MKReporterInternal.h"
+
+#import "MKResources.h"
+#import "MKUtil.hpp"
+
+MKUTIL_EXTEND_CLASS(MKReporterSettings, mk::collector::Settings);
+
+@implementation MKReporterSettings
+
+MKUTIL_INIT_WITH_IMPLICIT_CA_ASN_COUNTRY(
+  []() -> mk::collector::Settings *{
+    return new mk::collector::Settings;
+  }, [](mk::collector::Settings *p, const char *v) {
+    p->ca_bundle_path = v;
+  }, [](mk::collector::Settings *, const char *) {
+    // NOTHING
+  }, [](mk::collector::Settings *, const char *) {
+    // NOTHING
+  })
+
+MKUTIL_SET_STRING(setBaseURL, [](mk::collector::Settings *p, const char *v) {
+  p->base_url = v;
+})
+
+MKUTIL_SET_INT(setTimeout, [](mk::collector::Settings *p, int64_t v) {
+  p->timeout = v;
+})
+
+MKUTIL_DEINIT([](mk::collector::Settings *s) {
+  delete s; // handles nullptr gracefully
+})
+
+@end // @implementation MKReporterSettings
+
+@implementation MKReporterResults
+@end // @implementation MKReporterResults
+
+MKUTIL_EXTEND_CLASS(MKReporterTask, mk::collector::Reporter);
+
+@implementation MKReporterTask
+
+-(instancetype)initWithSettings:(MKReporterSettings *)settings
+                   softwareName:(NSString *)softwareName
+                softwareVersion:(NSString *)softwareVersion {
+  if (settings == nil || settings.impl == nil ||
+      softwareName == nil || softwareVersion == nil) {
+    abort();
+  }
+  if (([self init]) != nil) {
```

Comment:
Usually the conventional way to do this in Obj-C is
```
    self = [super init];
    if (self) {
//things
}
return self


```

Reply:
Will change the code, thanks for letting me know!